pkg/network/setup/configstate_test.go

```diff
@@ -0,0 +1,107 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package network_test
+
+import (
+	"errors"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	dutils "kubevirt.io/kubevirt/pkg/ephemeral-disk-utils"
+
+	"kubevirt.io/kubevirt/pkg/network/cache"
+	neterrors "kubevirt.io/kubevirt/pkg/network/errors"
+	network "kubevirt.io/kubevirt/pkg/network/setup"
+)
+
+const (
+	uid = "123"
+
+	testNet = "testnet"
+)
+
+var _ = Describe("config state", func() {
+	var (
+		configState      network.ConfigState
+		baseCacheCreator tempCacheCreator
+	)
+
+	BeforeEach(func() {
+		dutils.MockDefaultOwnershipManager()
+		configState = network.NewConfigState(&baseCacheCreator, uid)
+	})
+	AfterEach(func() {
+		Expect(baseCacheCreator.New("").Delete()).To(Succeed())
+	})
+
+	It("runs with no current state (cache is empty)", func() {
+		prepare, config := &funcStub{}, &funcStub{}
```

Comment:
Please don't call the discovery method `prepare` it is confusing since `prepare` is used to actually do the configuration in other place.

Reply:
Done